spring-cloud-gcp-spanner/src/test/java/org/springframework/data/spanner/core/mapping/BasicSpannerPersistentPropertyTests.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.data.spanner.core.mapping;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * @author Chengyuan Zhao
+ */
+@RunWith(SpringRunner.class)
+public class BasicSpannerPersistentPropertyTests {
+
+	@Test
+	public void testGetColumn() {
+		BasicSpannerPersistentEntity<TestEntity> entity =
+				(BasicSpannerPersistentEntity<TestEntity>) (new SpannerMappingContext()
+				.getPersistentEntity(TestEntity.class));
+
+		Set<String> cols = new HashSet<>();
+		cols.add("id");
+		cols.add("custom_col");
+		cols.add("other");
+
+		entity.columns().forEach(col -> {
+			BasicSpannerPersistentProperty prop = (BasicSpannerPersistentProperty) entity
+					.getPersistentPropertyByColumnName(col);
+			Assert.assertTrue(cols.contains(prop.getColumnName()));
+		});
+	}
+
+	@Test
+	public void testAssociations() {
+		BasicSpannerPersistentEntity<TestEntity> entity =
+				(BasicSpannerPersistentEntity<TestEntity>) (new SpannerMappingContext()
+				.getPersistentEntity(TestEntity.class));
+
+		entity.columns().forEach(col -> {
```

Comment:
Violation of single responsibility principle.
I'm not sure why we have to go through an existing Entity's properties. 
I'd rather just create a single property and check the sameness of the inverse and the "nullness" of the obverse.

Reply:
I definitely get what your'e saying, and I thought that too, but it's so awkward trying to rig up a property will all fake inputs and such. It was so convenient to use the mapping context to create the entity and then the properties.